src/test/java/org/broadinstitute/consent/http/resources/DatasetResourceTest.java

```diff
@@ -791,21 +795,26 @@ void testDownloadDatasetApprovedUsersError() {
   }
 
   @Test
-  void testFindAllDatasetsAvailableToUser() {
+  void testFindAllDatasetsStreaming() throws Exception {
+    var dataset = new Dataset();
+    dataset.setDataSetId(RandomUtils.nextInt(100, 1000));
     when(userService.findUserByEmail(any())).thenReturn(user);
-    when(datasetService.findAllDatasetsByUser(any())).thenReturn(List.of(new Dataset()));
+    final Gson gson = GsonUtil.buildGson();
+    StreamingOutput output = out -> out.write(gson.toJson(List.of(dataset)).getBytes());
+    when(datasetService.findAllDatasetsAsStreamingOutput()).thenReturn(output);
     initResource();
-    Response response = resource.findAllDatasetsAvailableToUser(authUser, null);
-    assertEquals(HttpStatusCodes.STATUS_CODE_OK, response.getStatus());
-  }
 
-  @Test
-  void testFindAllDatasetsAvailableToUserAsCustodian() {
-    when(userService.findUserByEmail(any())).thenReturn(user);
-    when(datasetService.findDatasetsByCustodian(any())).thenReturn(List.of(new Dataset()));
-    initResource();
-    Response response = resource.findAllDatasetsAvailableToUser(authUser, true);
+    Response response = resource.findAllDatasetsStreaming(authUser);
     assertEquals(HttpStatusCodes.STATUS_CODE_OK, response.getStatus());
+    var entity = (StreamingOutput) response.getEntity();
+    var baos = new ByteArrayOutputStream();
+    entity.write(baos);
+    var entityString = baos.toString();
+    Type listOfDatasetsType = new TypeToken<List<Dataset>>() {}.getType();
+    List<Dataset> returnedDatasets = gson.fromJson(entityString, listOfDatasetsType);
+    assertFalse(returnedDatasets.isEmpty());
+    assertEquals(1, returnedDatasets.size());
```

Comment:
These assertions could be consolidated like so:
```suggestion
    assertThat(returnedDatasets, hasSize(1));
```